src/sagas/FetchSaga.test.ts

```diff
@@ -0,0 +1,161 @@
+import { takeEvery, call, select } from "redux-saga/effects";
+import { describe } from 'riteway'
+
+import { Actions } from '../actions'
+import { fetchSaga, fetchData, dispatchFetchStatusUpdate, getLatestTwoNamesOnResource, apiFetch, TEXT, NOT_FOUND } from './FetchSaga'
+import { FetchType } from '../reducers/FetchReducer'
+import { getResourceState } from '../selectors/fetch'
+
+describe('FetchSaga()', async (should: any) => {
+  const { assert } = should()
+
+  const iterator = fetchSaga()();
+
+  assert({
+    given: 'fetchRequest Action',
+    should: 'handle fetch request',
+    actual: iterator.next().value,
+    expected: takeEvery(Actions.fetchRequest, fetchData),
+  })
+})
+
+describe('fetchData() Not Found Error', async (should: any) => {
+  const { assert } = should()
+  const url = { url: 'test'}
+  const short = getLatestTwoNamesOnResource(url.url)
+  const result: ReadonlyArray<string> = []
+  const error = NOT_FOUND
+  const headers = {} as Headers
+  const iterator = fetchData({ type: Actions.fetchRequest, payload: url })
+  
+  // REMOVE FOLLOWING LINE AFTER FIXING BELOW TEST
```

Comment:
@kennylavender @wzalazar @lautarodragan 
Would you know why this test is failing for me? It seems that they should both be returning that same value and the output looks to be the same, but it still fails.

Reply:
Is the test failing or is it that the tests, in general, are failing because of the bad import I commented on below?

Reply:
I've got the test passing with `expected: true and actual: true` right now. The lines that fail are commented out. It was running the tests with the lowercase f on my machine for some reason. I fixed the import though.

Reply:
Its because the selector is a curried function that takes the URL first. This means that each time you give it a URL, even if the URL is the same string, you get back a different function in memory. Even though the functions are equivalent, they are not the same function in memory so they will not equal each other.

`select` is not supposed to be used this way, you're supposed to give select a function and args to pass into the selector with the state.

```
select(getResourceState, url)
```

You will probably need to write the test in the correct format and fix the code.

Reply:
The MacOS filesystem is case insensitive by default, that's why it worked for you @jwicks31.

Reply:
![](https://media.giphy.com/media/wFOC9RazP97i0/giphy.gif)